drivers/sensor/bma400/bma400.c

```diff
@@ -0,0 +1,339 @@
+{
+	struct bma400_dev *drv_dev = dev->driver_data;
+	u8_t data[6];
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	ret = reg_read(BMA400_REG_ACC_X_LSB, data, 6, drv_dev);
+	if (ret == 0) {
+		drv_dev->data.x = convert_acc_data(data[0], data[1]);
+		drv_dev->data.y = convert_acc_data(data[2], data[3]);
+		drv_dev->data.z = convert_acc_data(data[4], data[5]);
+	} else   {
+		drv_dev->data.x = 0;
+		drv_dev->data.y = 0;
+		drv_dev->data.z = 0;
+	}
+
+	return ret;
+}
+
+int bma400_channel_get(struct device *dev, enum sensor_channel chan, 
+	struct sensor_value *val)
+{
+	struct bma400_dev *drv_dev = dev->driver_data;
+
+	if (chan == SENSOR_CHAN_ACCEL_X) {
+		channel_accel_convert(val, drv_dev->data.x, drv_dev->range);
+	} else if (chan == SENSOR_CHAN_ACCEL_Y) {
+		channel_accel_convert(val, drv_dev->data.y, drv_dev->range);
+	} else if (chan == SENSOR_CHAN_ACCEL_Z) {
+		channel_accel_convert(val, drv_dev->data.z, drv_dev->range);
+	} else if (chan == SENSOR_CHAN_ACCEL_XYZ) {
+		channel_accel_convert(&val[0], drv_dev->data.x, drv_dev->range);
+		channel_accel_convert(&val[1], drv_dev->data.y, drv_dev->range);
+		channel_accel_convert(&val[2], drv_dev->data.z, drv_dev->range);
+	} else   {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+int bma400_attr_set(struct device *dev, enum sensor_channel chan, 
+	enum sensor_attribute attr, const struct sensor_value *val)
+{
+	struct bma400_dev *drv_dev = dev->driver_data;
+	int ret;
+	double odr = drv_dev->last_odr;
+	double osr = drv_dev->last_osr;
+	double range = drv_dev->last_range;
+
+	if ((chan == SENSOR_CHAN_ACCEL_X) || (chan == SENSOR_CHAN_ACCEL_Y) || 
+		(chan == SENSOR_CHAN_ACCEL_Z) || (chan == SENSOR_CHAN_ACCEL_XYZ)) {
+		switch (attr) {
+		case SENSOR_ATTR_SAMPLING_FREQUENCY:
+			odr = (double) val->val1 + (double) val->val2 / 1000000;
+			drv_dev->last_odr = odr;
+			break;
+		case SENSOR_ATTR_OVERSAMPLING:
+			osr = (double) val->val1 + (double) val->val2 / 1000000;
+			drv_dev->last_osr = osr;
+			break;
+		case SENSOR_ATTR_FULL_SCALE:
+			range = (double) val->val1 + (double) val->val2 / 1000000;
+			drv_dev->last_range = range;
+			break;
+		default:
+			return -ENOTSUP;
+		}
+		ret = set_attributes(odr, osr, range, drv_dev);
+	} else   {
+		return -ENOTSUP;
+	}
+
+	return ret;
+}
+
+static int reg_read(u8_t reg, u8_t *data, u16_t length, 
+	struct bma400_dev *dev)
+{
+	int ret = 0;
+
+	if ((dev->i2c != NULL) && (data != NULL)) {
+		ret = i2c_burst_read(dev->i2c, dev->dev_id, reg, data, length);
+	} else   {
+		ret = -EINVAL;
+	}
+
+	return ret;
+}
+
+static int reg_write(u8_t reg, const u8_t *data, u16_t length, 
+	struct bma400_dev *dev)
+{
+	int ret = 0;
+	u16_t i;
+
+	if ((dev->i2c != NULL) && (data != NULL)) {
+		for (i = 0; i < length; i++) {
+			ret = i2c_burst_write(dev->i2c, dev->dev_id, reg + i, 
+				&data[i], 1);
+		}
+	} else   {
+		ret = -EINVAL;
+	}
+
+	return ret;
+}
+
+static void delay_us(u32_t period)
```

Comment:
remove delay_us and just use k_usleep directly.

Reply:
removed